src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs

```diff
@@ -1012,6 +1068,13 @@ private bool SyncTextAndValueProperties(bool updateValueFromText, string? text,
                 return null;
             }
             
+            if (TextConverter != null)
+            {
+                var valueFromText = TextConverter.Convert(text, typeof(decimal?), currentValueText, _cultureInfo ?? CultureInfo.CurrentUICulture);
+                if (valueFromText is decimal d)
```

Comment:
What exactly happens if converter returns a non-decimal value? The code seems to fall-through to the default parsing logic. Is that intended?

Reply:
Changed to direct cast. Let the user know immediately if he's doing the wrong thing.